scripts/pkg/update-docs.go

```diff
@@ -179,13 +185,28 @@ func fixupMarkdownFile(path string) error {
 	if DryRun {
 		return nil
 	}
+
+	// Prefix link to be added before relative links in
+	// order to translate relative links into absolute links.
+	prefix := fmt.Sprintf("(https://github.com/antrea-io/antrea/blob/%s/", version)
+
+	// Translate relative links to directories and files returned by getFilePrefixes() into absolute links.
+	relativeLinksExp := regexp.MustCompile(`\([./]*((` + strings.Join(getFilePrefixes(), `|`) + `)[\w.\-/]*\))`)
+	relativeLinks := relativeLinksExp.FindAll(md, -1)
+	for _, rl := range relativeLinks {
+		submatch := relativeLinksExp.FindSubmatch(rl)
+		suffix := string(submatch[1])
+		absLink := fmt.Sprintf("%s%s", prefix, suffix)
+		md = []byte(strings.Replace(string(md), string(rl), absLink, -1))
+	}
```

Comment:
I was suggesting doing it in one step with something like:
```go
relativeLinksExp := regexp.MustCompile(`\([./]*((build/|ci/|CHANGELOG)[\w.\-/]*)\)`)
relativeLinksExp.ReplaceAllString(md, "(https://github.com/antrea-io/antrea/blob/${1})")
```
see https://regex101.com/r/tlT2if/1 for the regex in action

Capture group 1 (`${1}`) is the part of the link that needs to be appended to the prefix.

Reply:
actually I have been thinking about this and I think we need a solution that's a bit more robust.

Imagine we have a sub-folder called `build/` somewhere in the antrea repo hierarchy (which is different from `/build`). Then the current logic may not work.
What we need instead:
- find all relative links (e.g., `../build/charts/antrea/crds/supportbundlecollection.yaml`)
- translate them into a relative link starting at the root of the directory (e.g., `../build/charts/antrea/crds/supportbundlecollection.yaml` => `build/charts/antrea/crds/supportbundlecollection.yaml`)
- prepend the absolute http link prefix

The second step is what is currently missing.

IMO, something like this may work well:
```go
re := <regex that matches all markdown links, excluding the ones that are URLs (start with `proto://`)>
mdDir := <relative path of the md file directory with respect to the root of the directory (e.g., 'docs' for 'docs/helm.md')>
re.ReplaceAllStringFunc(md, func(link string) string {
    newLink := link
    newLink = strings.TrimLeft(newLink, "(")
    newLink = strings.TrimRight(newLink, ")")
    // using path is more appropriate than path/filepath here
    if !path.IsAbs(newLink) {
        newLink = path.Join(mdDir, newLink)
    }
    newLink = path.Clean(newLink)
    for _, prefix := range prefixes {
        if strings.HasPrefix(newLink, prefix) {
            return fmt.Sprintf("(https://github.com/antrea-io/antrea/blob/%s/%s)", version, newLink)
        }
    }
    return link
})
```

Some variations to that code may be needed. This has also made me realize that we need to start adding unit tests for that code, as this is a pretty "complex" operation.

Reply:
That's a good point, and I had not thought about that. However, I think it's better to go with the more future-proof approach now. We could imagine having `docs/ci` at some point in the future. With the new approach, we will avoid bad surprises down the road. Please consider adding unit tests for these edge cases.

Reply:
done, PTAL.